docs/performance/inserts/parallel.txt

```diff
@@ -0,0 +1,131 @@
+.. _parallel_inserts:
+
+================
+Parallel Inserts
+================
+
+Because CrateDB is a clustered data with every node capable of handling client
+requests, parallelizing (i.e. doing many at once) your inserts is one good way
+to increase insert performance.
+
+There are a few ways you can do this.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Methods
+=======
+
+Use a Single Node
+-----------------
+
+Point your client application at a single node in your cluster, and send many
+concurrent requests.
+
+This is the least efficient way to do parallel inserts, because the
+resources on that node can easily become a bottleneck.
+
+Switch Between Multiple Nodes
+-----------------------------
+
+Configure your client application with a list of nodes in your cluster, and
+select a random node each time a request is made.
+
+The CrateDB `Python client`_, `JDBC client`_, and `PHP PDO client`_ support some
+sort of multiple node configuration.
+
+You can implement your own too.
+
+This solves the bottleneck issue when using a single node, but it's still a
+rudimentary approach that involves adding and removing node addresses manually
+from your application configuration.
+
+DNS Round Robin
+---------------
+
+When configuring your DNS, you can point the `A record`_ for a domain like
+``crate.example.com`` at multiple IP addresses. When a client requests the IP
+address for ``crate.example.com``, the DNS server will respond with a list of
+IP address, often in a random order. The client typically uses the first IP
+address in the list.
+
+This can be used as a sort of rudimentary load balancing system.
+
+There are a few problems:
+
+- The DNS does not know about CrateDB nodes that have joined or left the
+  cluster. You must update the configuration by hand each time.
+
+- DNS results are often cached by clients.
+
+  If you have a small number of clients, they will contact a small number of
+  nodes in your cluster.
+
+  DNS cache time (the *TTL* value) can be reduced, but this results in
+  additional DNS overhead, and doesn't help much if you have a large number of
+  nodes and small number of clients who are regularly stuck making requests to
+  a single node.
+
+  Ideally, clients should be rapidly switching beween CrateDB nodes, to make
+  best use of the cluster.
+
+Use a Load Balancer
+-------------------
+
+Stick your CrateDB cluster in front of a load balancer.
+
+With something like `HAProxy`_, you can configure a round robin load balancing
```

Comment:
"something" - you like this word :) I'd say "With a load balancer proxy like HaProxy"...